### ISSUE TITLE
Add prepack/postpack events to scripts

### ISSUE THREAD

**Issue by anurse** [dotnet/cli#242]
After #237, we have script execution in `dotnet`. We should add the `prepack` and `postpack` events which run before and after the `dotnet pack` command. Since `dotnet pack` executes `dotnet compile`, the flow looks like this:

1. `prepack`
2. `dotnet-compile`
  1. `precompile`
  2. `dotnet-compile-csc`
  3. `postcompile`
3. Prepare outputs for packaging
4. Create nupkg
5. `postpack`

/cc @davidfowl

**Comment by piotrpMSFT** [comment]
I think we should generalize this, per @davidfowl's suggestion.

@Sridhar-MS

**Comment by anurse** [comment]
I mostly agree, just want to keep this here temporarily to remember it until we finalize that plan. I'll close it out once we have a clean plan there

**Comment by piotrpMSFT** [comment]
This is a duplicate of #244
